library/user.inc

```diff
@@ -136,3 +136,34 @@ function prevSetting($uspfx, $postvar, $label, $initval)
 
     return $pset;
 }
+
+/**
+ * Function to set the state of expandable forms as per user choice, user default or global default
+ * @param array $filenames - array containing the names of the files being targeted, first element contains name of current file
+ * @param string $current_filename - derived from the first element of array $filenames
+ * @param integer $global_value - either the user defined or administrator defined global value
+ * @param integer $current_state - the current state of the form either 0 or 1
+ * @return the current state of the file after updating table user_settings
+ */
+
+function collectAndOrganizeExpandSetting($filenames = array())
+{
+    $current_filename = $filenames[0];
+    $global_value = $GLOBALS['expand_form'];
+      
+    if (getUserSetting($current_filename) > -1) {
+        $current_state = getUserSetting(attr($current_filename));
+    } elseif ($global_value) {
+        $current_state = $global_value;
+    } else {
+        $current_state = 0;
+    }
+   
+    if ($filenames.length) {
+        foreach ($filenames as $filename) {
+            setUserSetting(attr($filename), attr($current_state));
```

Comment:
remove the attr() function calls here. only should html escape stuff that goes out to the browser, or else we'll just twist our brains into a pretzel :)